### ISSUE TITLE
Relabeling default regex does not match newlines

### ISSUE THREAD

**Issue by roidelapluie** [prometheus/prometheus#8525]
The default regex for relabeling `(.*)` does not support newlines `(?s)(.*)`.

We can consider this a bug and fix it in the next minor release, or fix this in prometheus 3.

**Comment by alburthoffman** [comment]
it would be interesting that why would need new lines in a label value...

**Comment by roidelapluie** [comment]
The intention is that the default regex catches the new lines, but because of the widespread use of relabeling, we would prefer to delay this until prometheus 3.x.

**Comment by beorn7** [comment]
And once this is fixed, we can allow the `regex` field on relabeling in mode `hashmod`.

**Comment by beorn7** [comment]
Hello from the bug scrub.

To clarify: Once #7859 is done (which is planned for v3), the usual default regexp `(.*)` will implicitly also match newlines.

**Comment by jan--f** [comment]
I'll close this as a duplicate of #7859. Please reopen is there is additional work to be done.
